rewrite-java/src/main/java/org/openrewrite/java/AddNullMethodArgument.java

```diff
@@ -0,0 +1,125 @@
+ * which argument is added with a null value.
+ */
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AddNullMethodArgument extends Recipe {
+
+    /**
+     * A method pattern that is used to find matching method invocations.
+     * See {@link  MethodMatcher} for details on the expression's syntax.
+     */
+    @Option(displayName = "Method pattern",
+            description = "A method pattern that is used to find matching method invocations.",
+            example = "com.yourorg.A foo(int, int)")
+    String methodPattern;
+
+    /**
+     * A zero-based index that indicates which argument will be added as null to the method invocation.
+     */
+    @Option(displayName = "Argument index",
+            description = "A zero-based index that indicates which argument will be added as null to the method invocation.",
+            example = "0")
+    int argumentIndex;
+
+    @Override
+    public String getInstanceNameSuffix() {
+        return String.format("%d in methods `%s`", argumentIndex, methodPattern);
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Delete method argument";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Delete an argument from method invocations.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesMethod<>(methodPattern), new AddNullMethodArgumentVisitor(new MethodMatcher(methodPattern)));
+    }
+
+    private class AddNullMethodArgumentVisitor extends JavaIsoVisitor<ExecutionContext> {
+        private final MethodMatcher methodMatcher;
+
+        public AddNullMethodArgumentVisitor(MethodMatcher methodMatcher) {
+            this.methodMatcher = methodMatcher;
+        }
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+            J.MethodInvocation m = super.visitMethodInvocation(method, ctx);
+            return (J.MethodInvocation) visitMethodCall(m);
+        }
+
+        @Override
+        public J.NewClass visitNewClass(J.NewClass newClass, ExecutionContext ctx) {
+            J.NewClass n = super.visitNewClass(newClass, ctx);
+            return (J.NewClass) visitMethodCall(n);
+        }
+
+        private MethodCall visitMethodCall(MethodCall methodCall) {
+            MethodCall m = methodCall;
+            List<Expression> originalArgs = m.getArguments();
+            if (methodMatcher.matches(m) && (long) originalArgs.size() >= argumentIndex) {
+                List<Expression> args = new ArrayList<>(originalArgs);
+
+                if(args.size() == 1 && args.get(0) instanceof J.Empty) {
+                    args.remove(0);
+                }
+
+                args.add(argumentIndex, new J.Literal(randomId(), args.isEmpty() ? Space.EMPTY : Space.SINGLE_SPACE, Markers.EMPTY, "null", "null", null, JavaType.Primitive.Null));
+                m = m.withArguments(args);
```

Comment:
Perhaps this would be a little more in line with what we use elsewhere? 
```
m = m.withArguments(ListUtils.insert(m.getArguments(), new J.Literal(...), argumentIndex));
```

Reply:
I copied most of the cose (including this part) from `DeleteMethodArgument` but happy to change to this!